homework/hoanghai/lesson_1/202-HappyNumber.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @param {number} n
+ * @return {boolean}
+ */
+ var isHappy = function(n) {
+    // Array.includes() has a time complexity o(n)
+    // Set.has() has a time complexity o(1)
+    // apply set to prevent endless loop, and brute force solution to solve
+    if (n === 1) {
+        return true
+    }
+    let mnemonic = [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+    let preventEndlessLoop = new Set([0])
+    let sum = 0
+    while(!preventEndlessLoop.has(n)) {
```

Comment:
perhaps try Floyd's algorithm (slow and fast pointers) to avoid O(n) space of Set?